graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,219 @@
+    hh: 'Hamburg',
+    he: 'Hessen',
+    mv: 'Mecklenburg-Vorpommern',
+    ni: 'Niedersachsen',
+    nw: 'Nordrhein-Westfalen',
+    rp: 'Rheinland-Pfalz',
+    sl: 'Saarland',
+    sn: 'Sachsen',
+    st: 'Sachsen-Anhalt',
+    sh: 'Schleswig-Holstein',
+    th: 'Thüringen',
+    at: 'Austria',
+    ch: 'Switzerland',
+    other: 'Other',
+};
+
+/**
+ * Gets the full name of a state from its code.
+ * @param stateCode - The state code.
+ * @returns The full name of the state.
+ */
+export function getStateFullName(stateCode: StateCode): string {
+    return stateMap[stateCode] || 'Unknown';
+}
+
+@Resolver()
+export class MutateLessonPlanResolver {
+    @Mutation(() => LessonPlanOutput)
+    @Authorized(Role.INSTRUCTOR, Role.ADMIN, Role.STUDENT)
+    async generateLessonPlan(@Ctx() context: GraphQLContext, @Arg('data') data: GenerateLessonPlanInput): Promise<LessonPlanOutput> {
+        // Explicitly validate the input
+        try {
+            await validateOrReject(data);
+        } catch (errors) {
+            throw new UserInputError('Invalid input', { errors });
+        }
+
+        const { fileUuids, subject, grade, duration, state, prompt, expectedOutputs, schoolType, language } = data;
+
+        const fullStateName: string = getStateFullName(state);
+
+        try {
+            const generatedLessonPlan = await generateLessonPlan({
+                fileUuids: fileUuids || [],
+                subject,
+                grade,
+                duration,
+                state: fullStateName,
+                prompt,
+                expectedOutputs: expectedOutputs?.map((field) => field.toString()),
+                schoolType,
+                language,
+            });
+            logger.info(`Generated lesson plan for subject: ${subject}, grade: ${grade}, school type: ${schoolType}, language: ${language || 'German'}`);
+
+            // Create the lesson plan output based on the expected outputs
+            const lessonPlan: LessonPlanOutput = {
+                subject: generatedLessonPlan.subject,
+                grade: generatedLessonPlan.grade,
+                duration: generatedLessonPlan.duration,
+            };
+
+            // Only include the fields that were requested or generated
+            if (expectedOutputs) {
+                for (const field of expectedOutputs) {
+                    if (field in generatedLessonPlan) {
+                        lessonPlan[field] = generatedLessonPlan[field];
+                    }
+                }
+            } else {
+                // If no expected outputs were specified, include all generated fields
+                Object.assign(lessonPlan, generatedLessonPlan);
+            }
+
+            return lessonPlan;
+        } catch (error) {
+            logger.error(`Error generating lesson plan: ${error.message}`);
+
+            if (error.message.includes('Invalid fileID')) {
+                throw new UserInputError('Invalid file UUID provided', {
```

Comment:
In https://github.com/corona-school/backend/blob/master/common/util/error.ts we already define errors that can be used in common/ code, and which when thrown get exposed to the GraphQL client (unlike other errors which are shown as "Internal Server Errors") - I would prefer to throw these in common/lessonplan instead in all cases where we expect a user error - then we can avoid catching + rethrowing here and do not need the LessonPlanGenerationError ...